examples/connectivity/compare_resting_state_decomposition.py

```diff
@@ -0,0 +1,81 @@
+"""
+Comparison of Dictionary Learning and ICA for doing group analysis of
+resting-state fMRI
+=====================================================
+
+An example applying dictionary learning and ICA to resting-state data,
+comparing resulting components using 4D plotting.
+
+Dictionary learning is a sparsity based decomposition method for extracting
+spatial maps. It extracts maps that are naturally sparse and usually cleaner
+than ICA
+
+    * Gael Varoquaux et al.
+    Multi-subject dictionary learning to segment an atlas of brain spontaneous
+    activity
+    Information Processing in Medical Imaging, 2011, pp. 562-573, Lecture Notes
+    in Computer Science
+
+https://hal.inria.fr/inria-00588898/en/
+"""
+
+### Load ADHD rest dataset ####################################################
+from nilearn import datasets
+
+adhd_dataset = datasets.fetch_adhd(n_subjects=40)
+func_filenames = adhd_dataset.func  # list of 4D nifti files for each subject
+
+# print basic information on the dataset
+print('First functional nifti image (4D) is at: %s' %
+      adhd_dataset.func[0])  # 4D data
+
+### Apply Decomposition estimators#############################################
+from nilearn.decomposition import DictLearning, CanICA
+
+n_components = 20
+
+### Dictionary learning #######################################################
+dict_learning = DictLearning(n_components=n_components, smoothing_fwhm=6.,
+                             memory="nilearn_cache", memory_level=2,
+                             verbose=1,
+                             alpha=2,
+                             random_state=0,
+                             n_epochs=1)
+### CanICA ####################################################################
+canica = CanICA(n_components=n_components, smoothing_fwhm=6.,
+                memory="nilearn_cache",  memory_level=2,
+                threshold=3.,
+                verbose=1)
+
+### Fitting both estimators ###################################################
+estimators = [dict_learning, canica]
+components_imgs = []
+
+for estimator in estimators:
+    print('[Example] Learning maps using %s model' % type(estimator).__name__)
+    estimator.fit(func_filenames)
+    print('[Example] Dumping results')
+    # Decomposition estimator embeds their own masker
+    masker = estimator.masker_
+    components_img = masker.inverse_transform(estimator.components_)
+    components_img.to_filename('%s_resting_state.nii.gz' %
```

Comment:
> Do you actually need to save components to file?

I think that it is useful because users will most likely want to do that,
and that way, when they copy-paste from the example they don't have to
look for how to do it.